internal/provider/providers/vultr/provider.go

```diff
@@ -0,0 +1,190 @@
+package vultr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/headers"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	domain     string
+	owner      string
+	ipVersion  ipversion.IPVersion
+	ipv6Suffix netip.Prefix
+	apiKey      string
+	ttl        uint32
+}
+
+func New(data json.RawMessage, domain, owner string,
+	ipVersion ipversion.IPVersion, ipv6Suffix netip.Prefix) (
+	provider *Provider, err error) {
+	var providerSpecificSettings struct {
+		APIKey string `json:"apiKey"`
```

Comment:
```suggestion
		APIKey string `json:"apikey"`
```

Reply:
Sure, but why not camelCase? 🤔

Reply:
Just to match other existing fields, which are all snake case. Also a bit of a habit to have snake case json keys.